jib-core/src/test/java/com/google/cloud/tools/jib/http/TestWebServer.java

```diff
@@ -119,14 +119,18 @@ private Void serveResponses() throws IOException {
         out.write("HTTP/1.1 400 Bad Request\n\n".getBytes(StandardCharsets.UTF_8));
         return null;
       }
-      inputRead.append((char) firstByte);
 
       BufferedReader reader = new BufferedReader(new InputStreamReader(in, StandardCharsets.UTF_8));
       for (String response : responses) {
         for (String line = reader.readLine();
             line != null && !line.isEmpty(); // An empty line marks the end of an HTTP request.
             line = reader.readLine()) {
-          inputRead.append(line + "\n");
+          if (firstByte == -1) {
+            inputRead.append(line + "\n");
+          } else {
+            inputRead.append(((char) firstByte) + line + "\n");
+            firstByte = -1;
+          }
```

Comment:
Could you explain this change? I'm confused about why it's necessary.

Reply:
I wanted to ensure lines are never cut in the middle. For example,

Thread 1: does `inputRead.append('G')
Thread 2: does `inputRead.append('G')
Thread 1 & 2: read the rest of their respective line and do `input.append(line + "\n")`

Then `inputRead` will have the content:

 ```
GGET /path HTTP...
ET /path HTTP...
```

I just wanted to prevent that.